src/workerd/api/tests/js-rpc-test.js

```diff
@@ -1179,11 +1197,40 @@ export let crossContextSharingDoesntWork = {
 
 export let waitUntilWorks = {
   async test(controller, env, ctx) {
-    globalWaitUntilPromise = null;
-    await env.MyService.testWaitUntil();
+    // Tests ctx.waitUntil
+    {
+      globalWaitUntilPromise = null;
+      await env.MyService.testWaitUntil();
+
+      assert.strictEqual(globalWaitUntilPromise instanceof Promise, true);
+      await globalWaitUntilPromise;
+    }
+
+    // Tests `import { waitUntil } from 'cloudflare:workers` on DO
```

Comment:
I'm confused, the code below does not test on a DO AFAICT?

But I don't think such a test is worth it anyway.

Reply:
Yes, my comment was wrong. Updating it.